src/firmware/linux/ioctl/guest.rs

```diff
@@ -0,0 +1,60 @@
+use std::marker::PhantomData;
+
+use crate::firmware::types::*;
+use crate::impl_const_id;
+
+use iocuddle::*;
+
+// These enum ordinal values are defined in the Linux kernel
+// source code: include/uapi/linux/psp-sev.h
+impl_const_id! {
+    pub Id => u32;
+    GuestMsg<'_, '_> = 0,
+    DerivedKey = 1,
+    ExtReport = 2,
+}
+
+const SEV: Group = Group::new(b'S');
+
+pub const GET_REPORT: Ioctl<WriteRead, &Command<GuestMsg>> = unsafe { SEV.write_read(0) };
+pub const GET_DERIVED_KEY: Ioctl<WriteRead, &Command<DerivedKey>> = unsafe { SEV.write_read(0) };
+pub const GET_EXT_REPORT: Ioctl<WriteRead, &Command<ExtReport>> = unsafe { SEV.write_read(0) };
```

Comment:
I think the last two ioctl numbers are wrong:

```suggestion
pub const GET_DERIVED_KEY: Ioctl<WriteRead, &Command<DerivedKey>> = unsafe { SEV.write_read(1) };
pub const GET_EXT_REPORT: Ioctl<WriteRead, &Command<ExtReport>> = unsafe { SEV.write_read(2) };
```